src/PluginsSystem/Tools/Microsoft.Performance.Toolkit.Plugins.Cli/Processing/PluginArtifactsProcessor.cs

```diff
@@ -109,10 +109,11 @@ public bool TryProcess(PluginArtifacts artifacts, [NotNullWhen(true)] out Proces
                     {
                         assembly = Assembly.LoadFrom(file);
```

Comment:
edit: I see we use it below... why do we call this here?

Reply:
The PluginsLoader doesn't expose the SDK version of the plugin in the loading result. It does validate the version using VersionChecker but doesn't save the version. We probably should add a TODO to make pluginsloader save that. It's not trivial.

Reply:
Hmm this is something we're going to have to do sooner rather than later. We can't just bail on these files, because e.g. the tool can be ran on an x64 machine for an arm64 plugin (although, plugins should _really_ be compiled for AnyCPU...).

We use [MetadataLoadContext](https://learn.microsoft.com/en-us/dotnet/standard/assembly/inspect-contents-using-metadataloadcontext) in the runtime to inspect the referenced SDK without loading the assembly. It's built for this type of scenario. Maybe we need to use that here? Or figure out a way to use the runtime to get this info?

Reply:
Tested `MetadataLoadContext.Load` - it will still throw if when loading native dlls.